web/src/features/map/map-layers/GridAlertsLayer.tsx

```diff
@@ -0,0 +1,131 @@
+import useGetState from 'api/getState';
+import { getTextColor } from 'components/CarbonIntensitySquare';
+import { useCo2ColorScale } from 'hooks/theme';
+import { useAtomValue } from 'jotai';
+import { useMemo } from 'react';
+import { FiAlertTriangle } from 'react-icons/fi';
+import { Marker } from 'react-map-gl/maplibre';
+import { GridState, MapGeometries } from 'types';
+import { getCarbonIntensity } from 'utils/helpers';
+import { isConsumptionAtom, selectedDatetimeStringAtom } from 'utils/state/atoms';
+
+import { useGetGeometries } from '../map-utils/getMapGrid';
+
+// Helper function to calculate the centroid of a polygon
+const getPolygonCentroid = (coordinates: number[][][]) => {
+  let x = 0,
+    y = 0,
+    area = 0;
+  const ring = coordinates[0]; // Use outer ring
+
+  for (let index = 0; index < ring.length - 1; index++) {
+    const xi = ring[index][0];
+    const yi = ring[index][1];
+    const xi1 = ring[index + 1][0];
+    const yi1 = ring[index + 1][1];
+    const a = xi * yi1 - xi1 * yi;
+    area += a;
+    x += (xi + xi1) * a;
+    y += (yi + yi1) * a;
+  }
+
+  area *= 0.5;
+  x /= 6 * area;
+  y /= 6 * area;
+
+  return [x, y];
+};
```

Comment:
There are helpers from turf that can do this if we wish (same library we use to compile the map) but this works!